cpp/src/arrow/csv/column_decoder.cc

```diff
@@ -52,10 +51,12 @@ class ConcreteColumnDecoder : public ColumnDecoder {
   // XXX useful?
   virtual std::shared_ptr<DataType> type() const = 0;
 
-  Status WrapConversionError(const Status& st) {
-    if (st.ok()) {
-      return st;
+  inline Result<std::shared_ptr<Array>> WrapConversionError(
```

Comment:
No need to add `inline` here, it is implied when a method definition is inside the `class` declaration.

Reply:
Removed